Rax Roles Extension requires upper case in tenanted role parameter names.

Given the WADL below:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<application xmlns="http://wadl.dev.java.net/2009/02"
             xmlns:rax="http://docs.rackspace.com/api"
             xmlns:xsd="http://www.w3.org/2001/XMLSchema">
    <resources base="https://test.api.openstack.com">
        <resource path="/v1/resource" rax:roles="a:admin/{X-TENANT-ID}">
            <param name="X-OTHER"
                   style="header"
                   required="true"
                   type="xsd:string"
                   repeating="false"/>
            <method name="POST" rax:roles="a:creator/{x-tenant-id}">
                <request>
                    <param name="x-tenant-id"
                           style="header"
                           required="true"
                           type="xsd:int"
                           repeating="true"
                           rax:isTenant="true"/>
                    <param name="X-Tenant-Id"
                           style="header"
                           required="true"
                           type="xsd:boolean"
                           repeating="true"
                           rax:isTenant="true"/>
                </request>
            </method>
            <method name="GET" rax:roles="a:observer/{X-TENANT-ID}">
                <request>
                    <param name="x-tENANT-iD"
                           style="header"
                           required="true"
                           type="xsd:int"
                           repeating="true"
                           rax:isTenant="true"/>
                    <param name="X-TENANT-ID"
                           style="header"
                           required="true"
                           type="xsd:boolean"
                           repeating="true"
                           rax:isTenant="true"/>
                </request>
            </method>
            <method name="PUT" rax:roles="a:updater/{X-TENANT-ID}">
                <request>
                    <param name="x-tenant-id"
                           style="header"
                           required="true"
                           type="xsd:int"
                           repeating="true"
                           rax:isTenant="true"/>
                    <param name="X-TENANT-ID"
                           style="header"
                           required="true"
                           type="xsd:boolean"
                           repeating="true"
                           rax:isTenant="true"/>
                </request>
            </method>
            <method name="DELETE">
                <request>
                    <param name="X-Tenant-Id"
                           style="header"
                           required="true"
                           type="xsd:int"
                           repeating="true"
                           rax:isTenant="true"/>
                    <param name="x-tENANT-iD"
                           style="header"
                           required="true"
                           type="xsd:boolean"
                           repeating="true"
                           rax:isTenant="true"/>
                </request>
            </method>
            <resource path="other">
                <method name="POST" rax:roles="a:creator/{X-TENANT-ID}">
                    <request>
                        <param name="x-tenant-id"
                               style="header"
                               required="true"
                               type="xsd:int"
                               repeating="true"
                               rax:isTenant="true"/>
                        <param name="x-tENANT-iD"
                               style="header"
                               required="true"
                               type="xsd:boolean"
                               repeating="true"
                               rax:isTenant="true"/>
                    </request>
                </method>
                <method name="GET" rax:roles="#all">
                    <request>
                        <param name="X-Tenant-Id"
                               style="header"
                               required="true"
                               type="xsd:int"
                               repeating="true"
                               rax:isTenant="true"/>
                        <param name="X-TENANT-ID"
                               style="header"
                               required="true"
                               type="xsd:boolean"
                               repeating="true"
                               rax:isTenant="true"/>
                    </request>
                </method>
                <method name="PUT" rax:roles="a:updater/{X-TENANT-ID}">
                    <request>
                        <param name="X-TENANT-ID"
                               style="header"
                               required="true"
                               type="xsd:int"
                               repeating="true"
                               rax:isTenant="true"/>
                        <param name="X-TENANT-ID"
                               style="header"
                               required="true"
                               type="xsd:boolean"
                               repeating="true"
                               rax:isTenant="true"/>
                    </request>
                </method>
                <method name="DELETE">
                    <request>
                        <param name="X-TENANT-ID"
                               style="header"
                               required="true"
                               type="xsd:int"
                               repeating="true"
                               rax:isTenant="true"/>
                        <param name="X-TENANT-ID"
                               style="header"
                               required="true"
                               type="xsd:boolean"
                               repeating="true"
                               rax:isTenant="true"/>
                    </request>
                </method>
                <method name="PATCH" rax:roles="role&#xA0;with&#xA0;spaces/{X-TENANT-ID} a:patcher/{X-TENANT-ID}">
                    <request>
                        <param name="X-TENANT-ID"
                               style="header"
                               required="true"
                               type="xsd:int"
                               repeating="true"
                               rax:isTenant="true"/>
                        <param name="X-TENANT-ID"
                               style="header"
                               required="true"
                               type="xsd:boolean"
                               repeating="true"
                               rax:isTenant="true"/>
                    </request>
                </method>
            </resource>
        </resource>
    </resources>
</application>
```

This is the stacktrace received:
```
2018-02-19 17:07:47,415 18871 [Event Kernel Thread] ERROR com.rackspace.cloud.api.wadl.util.LogErrorListener - rax:roles match for role 'a:observer/{x-tenant-id}', but no defined param named 'x-tenant-id' in this case.
2018-02-19 17:07:47,417 18873 [Event Kernel Thread] ERROR com.rackspace.cloud.api.wadl.util.LogErrorListener - Processing terminated by xsl:message at line 423 in raxRolesTenants.xsl
2018-02-19 17:07:47,418 18874 [Event Kernel Thread] ERROR com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder - Processing terminated by xsl:message at line 423 in raxRolesTenants.xsl
2018-02-19 17:07:47,418 18874 [Event Kernel Thread] WARN  org.openrepose.filters.apivalidator.ValidatorInfo - Error loading validator for WADL: file:/home/bill7601/Projects/repose/repose-aggregator/tests/functional-tests/build/test-homes/features/filters/apivalidator/ApiValidatorMultiTenantTest/configs/headerAllAtMethodTenantExplicit.wadl.xml
com.rackspace.com.papi.components.checker.wadl.WADLException: WADL Processing Error: Processing terminated by xsl:message at line 423 in raxRolesTenants.xsl
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder.buildFromWADL(WADLCheckerBuilder.scala:749) ~[checker-core-2.6.1-SNAPSHOT.jar:2.6.1-SNAPSHOT]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder.build(WADLCheckerBuilder.scala:808) ~[checker-core-2.6.1-SNAPSHOT.jar:2.6.1-SNAPSHOT]
	at com.rackspace.com.papi.components.checker.wadl.StepBuilder.build(StepBuilder.scala:59) ~[checker-core-2.6.1-SNAPSHOT.jar:2.6.1-SNAPSHOT]
	at com.rackspace.com.papi.components.checker.Validator$.apply(Validator.scala:67) ~[checker-core-2.6.1-SNAPSHOT.jar:2.6.1-SNAPSHOT]
	at com.rackspace.com.papi.components.checker.Validator.apply(Validator.scala) ~[checker-core-2.6.1-SNAPSHOT.jar:2.6.1-SNAPSHOT]
	at org.openrepose.filters.apivalidator.ValidatorInfo.initValidator(ValidatorInfo.java:96) [api-validator-filter-8.8.1.0-SNAPSHOT.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.filters.apivalidator.ValidatorInfo.reinitValidator(ValidatorInfo.java:122) [api-validator-filter-8.8.1.0-SNAPSHOT.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.filters.apivalidator.ApiValidatorHandlerFactory$ApiValidatorWadlListener.configurationUpdated(ApiValidatorHandlerFactory.java:188) [api-validator-filter-8.8.1.0-SNAPSHOT.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.filters.apivalidator.ApiValidatorHandlerFactory$ApiValidatorWadlListener.configurationUpdated(ApiValidatorHandlerFactory.java:160) [api-validator-filter-8.8.1.0-SNAPSHOT.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.core.services.config.impl.ConfigurationServiceImpl.subscribeTo(ConfigurationServiceImpl.java:137) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.core.services.config.impl.ConfigurationServiceImpl.subscribeTo(ConfigurationServiceImpl.java:128) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.filters.apivalidator.ApiValidatorHandlerFactory.addWadlListener(ApiValidatorHandlerFactory.java:126) [api-validator-filter-8.8.1.0-SNAPSHOT.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.filters.apivalidator.ApiValidatorHandlerFactory.initialize(ApiValidatorHandlerFactory.java:113) [api-validator-filter-8.8.1.0-SNAPSHOT.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.filters.apivalidator.ApiValidatorHandlerFactory.configurationUpdated(ApiValidatorHandlerFactory.java:88) [api-validator-filter-8.8.1.0-SNAPSHOT.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.filters.apivalidator.ApiValidatorHandlerFactory.configurationUpdated(ApiValidatorHandlerFactory.java:45) [api-validator-filter-8.8.1.0-SNAPSHOT.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.core.services.config.impl.ConfigurationServiceImpl.subscribeTo(ConfigurationServiceImpl.java:137) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.core.services.config.impl.ConfigurationServiceImpl.subscribeTo(ConfigurationServiceImpl.java:120) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.filters.apivalidator.ApiValidatorFilter.init(ApiValidatorFilter.java:70) [api-validator-filter-8.8.1.0-SNAPSHOT.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.powerfilter.filtercontext.FilterContextFactory.loadFilterContext(FilterContextFactory.java:123) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.powerfilter.filtercontext.FilterContextFactory.buildFilterContexts(FilterContextFactory.java:63) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.powerfilter.PowerFilter.configurationHeartbeat(PowerFilter.java:225) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.powerfilter.PowerFilter.access$600(PowerFilter.java:91) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.powerfilter.PowerFilter$ApplicationDeploymentEventListener.onEvent(PowerFilter.java:479) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.core.services.event.EventDispatcherImpl.dispatch(EventDispatcherImpl.java:43) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at org.openrepose.core.services.event.PowerProxyEventKernel.run(PowerProxyEventKernel.java:75) [repose-valve.jar:8.8.1.0-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]
Caused by: net.sf.saxon.s9api.SaxonApiException: Processing terminated by xsl:message at line 423 in raxRolesTenants.xsl
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:606) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder.com$rackspace$com$papi$components$checker$wadl$WADLCheckerBuilder$$raxRolesTenants(WADLCheckerBuilder.scala:502) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1$$anonfun$7.apply(WADLCheckerBuilder.scala:736) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1$$anonfun$7.apply(WADLCheckerBuilder.scala:736) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.BuilderHelper$.applyBuildSteps(BuilderHelper.scala:129) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1.apply$mcV$sp(WADLCheckerBuilder.scala:744) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1.apply(WADLCheckerBuilder.scala:703) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1.apply(WADLCheckerBuilder.scala:703) ~[?:?]
	at com.rackspace.cloud.api.wadl.util.XSLErrorDispatcher$class.handleXSLException(xsl-error-dispatcher.scala:15) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder.handleXSLException(WADLCheckerBuilder.scala:233) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder.buildFromWADL(WADLCheckerBuilder.scala:703) ~[?:?]
	... 25 more
Caused by: net.sf.saxon.expr.instruct.TerminationException: Processing terminated by xsl:message at line 423 in raxRolesTenants.xsl
	at net.sf.saxon.expr.instruct.Message.processLeavingTail(Message.java:223) ~[?:?]
	at net.sf.saxon.expr.instruct.Instruction.process(Instruction.java:151) ~[?:?]
	at net.sf.saxon.expr.parser.ExpressionTool.evaluate(ExpressionTool.java:350) ~[?:?]
	at net.sf.saxon.expr.parser.ExpressionTool.eagerEvaluate(ExpressionTool.java:463) ~[?:?]
	at net.sf.saxon.expr.TryCatch.iterate(TryCatch.java:281) ~[?:?]
	at net.sf.saxon.expr.parser.ExpressionTool.evaluate(ExpressionTool.java:340) ~[?:?]
	at net.sf.saxon.expr.parser.ExpressionTool.eagerEvaluate(ExpressionTool.java:463) ~[?:?]
	at net.sf.saxon.expr.TryCatch.iterate(TryCatch.java:281) ~[?:?]
	at net.sf.saxon.expr.Expression.process(Expression.java:927) ~[?:?]
	at net.sf.saxon.expr.LetExpression.processLeavingTail(LetExpression.java:703) ~[?:?]
	at net.sf.saxon.expr.instruct.TemplateRule.applyLeavingTail(TemplateRule.java:347) ~[?:?]
	at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:502) ~[?:?]
	at net.sf.saxon.trans.TextOnlyCopyRuleSet.process(TextOnlyCopyRuleSet.java:69) ~[?:?]
	at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:478) ~[?:?]
	at net.sf.saxon.trans.TextOnlyCopyRuleSet.process(TextOnlyCopyRuleSet.java:69) ~[?:?]
	at net.sf.saxon.trans.Mode.applyTemplates(Mode.java:478) ~[?:?]
	at net.sf.saxon.expr.instruct.ApplyTemplates.apply(ApplyTemplates.java:295) ~[?:?]
	at net.sf.saxon.expr.instruct.ApplyTemplates.process(ApplyTemplates.java:248) ~[?:?]
	at net.sf.saxon.expr.parser.ExpressionTool.getIteratorFromProcessMethod(ExpressionTool.java:799) ~[?:?]
	at net.sf.saxon.expr.instruct.Instruction.iterate(Instruction.java:379) ~[?:?]
	at net.sf.saxon.expr.ItemChecker.iterate(ItemChecker.java:227) ~[?:?]
	at com.saxonica.ee.bytecode.ByteCodeCandidate.iterate(ByteCodeCandidate.java:290) ~[?:?]
	at net.sf.saxon.value.Closure.iterate(Closure.java:156) ~[?:?]
	at net.sf.saxon.expr.UserFunctionCall.iterate(UserFunctionCall.java:487) ~[?:?]
	at net.sf.saxon.expr.parser.Evaluator$5.evaluate(Evaluator.java:101) ~[?:?]
	at net.sf.saxon.expr.SystemFunctionCall.evaluateArguments(SystemFunctionCall.java:448) ~[?:?]
	at net.sf.saxon.expr.FunctionCall.iterate(FunctionCall.java:545) ~[?:?]
	at net.sf.saxon.expr.parser.ExpressionTool.evaluate(ExpressionTool.java:340) ~[?:?]
	at net.sf.saxon.expr.LetExpression.eval(LetExpression.java:511) ~[?:?]
	at net.sf.saxon.expr.LetExpression.iterate(LetExpression.java:485) ~[?:?]
	at com.saxonica.ee.bytecode.ByteCodeCandidate.iterate(ByteCodeCandidate.java:290) ~[?:?]
	at net.sf.saxon.expr.parser.ExpressionTool.evaluate(ExpressionTool.java:340) ~[?:?]
	at net.sf.saxon.expr.instruct.UserFunction.call(UserFunction.java:616) ~[?:?]
	at net.sf.saxon.expr.UserFunctionCall.callFunction(UserFunctionCall.java:541) ~[?:?]
	at net.sf.saxon.expr.UserFunctionCall.evaluateItem(UserFunctionCall.java:477) ~[?:?]
	at net.sf.saxon.expr.parser.ExpressionTool.evaluate(ExpressionTool.java:328) ~[?:?]
	at net.sf.saxon.expr.LetExpression.eval(LetExpression.java:511) ~[?:?]
	at net.sf.saxon.expr.LetExpression.process(LetExpression.java:580) ~[?:?]
	at com.saxonica.ee.bytecode.ByteCodeCandidate.process(ByteCodeCandidate.java:141) ~[?:?]
	at net.sf.saxon.expr.instruct.NamedTemplate.expand(NamedTemplate.java:264) ~[?:?]
	at net.sf.saxon.expr.instruct.CallTemplate$CallTemplatePackage.processLeavingTail(CallTemplate.java:518) ~[?:?]
	at net.sf.saxon.Controller.transformDocument(Controller.java:2404) ~[?:?]
	at net.sf.saxon.Controller.transform(Controller.java:1970) ~[?:?]
	at net.sf.saxon.s9api.XsltTransformer.transform(XsltTransformer.java:596) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder.com$rackspace$com$papi$components$checker$wadl$WADLCheckerBuilder$$raxRolesTenants(WADLCheckerBuilder.scala:502) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1$$anonfun$7.apply(WADLCheckerBuilder.scala:736) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1$$anonfun$7.apply(WADLCheckerBuilder.scala:736) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.BuilderHelper$.applyBuildSteps(BuilderHelper.scala:129) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1.apply$mcV$sp(WADLCheckerBuilder.scala:744) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1.apply(WADLCheckerBuilder.scala:703) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder$$anonfun$buildFromWADL$1.apply(WADLCheckerBuilder.scala:703) ~[?:?]
	at com.rackspace.cloud.api.wadl.util.XSLErrorDispatcher$class.handleXSLException(xsl-error-dispatcher.scala:15) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder.handleXSLException(WADLCheckerBuilder.scala:233) ~[?:?]
	at com.rackspace.com.papi.components.checker.wadl.WADLCheckerBuilder.buildFromWADL(WADLCheckerBuilder.scala:703) ~[?:?]
	... 25 more
```

It is being caused by the value of the first `method` element's `rax:roles` attribute. Regardless of the case of the `param` element's `name` attribute, the corresponding value in the `rax:roles` attribute currently must upper case. Any other permutation of the value causes a similar stacktrace.

--- a/RELEASE.md
+++ b/RELEASE.md
@@ -2,6 +2,7 @@
 ## In Progress Work ##
 1. Fixed a bug where adding extra whitespace in the ```rax:roles``` attribute caused errors.
 1. Fixed a bug where we were not reporting a proper error if a method name was missing.
+1. Fixed a bug where the case of a HEADER_ALL parameter name was not being preserved.
 1. Updated Dependencies
    1. saxon: 9.8.0-4 → 9.8.0-8
    1. wadl-tools: 1.0.37 → 1.0.38

--- a/core/src/main/resources/xsl/builder.xsl
+++ b/core/src/main/resources/xsl/builder.xsl
@@ -830,7 +830,7 @@
                   </xsl:if>
                 </xsl:for-each>
               </xsl:variable>
-              <step type="HEADER_ALL" id="{check:HeaderID($allHeaders[1])}" name="{$current}">
+              <step type="HEADER_ALL" id="{check:HeaderID($allHeaders[1])}" name="{$allHeaders[1]/@name}">
                 <xsl:if test="not(empty($matches))">
                   <xsl:attribute name="matchRegEx"><xsl:value-of select="distinct-values($matches)" separator="|"/></xsl:attribute>
                 </xsl:if>

--- a/core/src/test/scala/com/rackspace/com/papi/components/checker/wadl/WADLCheckerTenantFailSpec.scala
+++ b/core/src/test/scala/com/rackspace/com/papi/components/checker/wadl/WADLCheckerTenantFailSpec.scala
@@ -36,6 +36,7 @@ class WADLCheckerTenantFailSpec extends BaseCheckerSpec with LogAssertions {
     val c = new Config()
     c.enableRaxRolesExtension = true
     c.maskRaxRoles403 = true
+    c.checkPlainParams = true
     c
   }
 
@@ -43,6 +44,7 @@ class WADLCheckerTenantFailSpec extends BaseCheckerSpec with LogAssertions {
     val c = new Config()
     c.enableRaxRolesExtension = true
     c.maskRaxRoles403 = false
+    c.checkPlainParams = true
     c
   }
 
@@ -138,6 +140,331 @@ class WADLCheckerTenantFailSpec extends BaseCheckerSpec with LogAssertions {
       Then ("There should be an error detailing that rax:roles mask is currently not supported")
       assert(checkerLog, "no defined param named 'foo'")
     }
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (uri)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/{id}/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="id" style="template" type="xsd:string"/>
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#getMethod" />
+              </resource>
+           </resources>
+           <method id="getMethod" name="GET">
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (xpath)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#postMethod" />
+              </resource>
+           </resources>
+           <method id="postMethod" name="POST">
+               <request>
+                 <representation mediaType="application/xml">
+                    <param name="id" style="plain" path="//@id" required="true"/>
+                 </representation>
+               </request>
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (xpath-json)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#postMethod" />
+              </resource>
+           </resources>
+           <method id="postMethod" name="POST">
+               <request>
+                 <representation mediaType="application/json">
+                    <param name="id" style="plain" path="$body(@id)" required="true"/>
+                 </representation>
+               </request>
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (header)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="id" style="header" repeating="true" required="true"/>
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#postMethod" />
+              </resource>
+           </resources>
+           <method id="postMethod" name="POST">
+               <request>
+                 <representation mediaType="application/json">
+                 </representation>
+               </request>
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (header any)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="id" style="header" repeating="true" required="true" rax:anyMatch="true" fixed="foo"/>
+                   <param name="id" style="header" repeating="true" required="true" rax:anyMatch="true" fixed="bar"/>
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#postMethod" />
+              </resource>
+           </resources>
+           <method id="postMethod" name="POST">
+               <request>
+                 <representation mediaType="application/json">
+                 </representation>
+               </request>
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (header any, first match)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="id" style="header" repeating="true" required="true" rax:anyMatch="true" fixed="foo"/>
+                   <param name="ID" style="header" repeating="true" required="true" rax:anyMatch="true" fixed="bar"/>
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#postMethod" />
+              </resource>
+           </resources>
+           <method id="postMethod" name="POST">
+               <request>
+                 <representation mediaType="application/json">
+                 </representation>
+               </request>
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (header all)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="id" style="header" repeating="true" required="true" fixed="foo"/>
+                   <param name="id" style="header" repeating="true" required="true" fixed="bar"/>
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#postMethod" />
+              </resource>
+           </resources>
+           <method id="postMethod" name="POST">
+               <request>
+                 <representation mediaType="application/json">
+                 </representation>
+               </request>
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (header all, first match)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="id" style="header" repeating="true" required="true" fixed="foo"/>
+                   <param name="ID" style="header" repeating="true" required="true" fixed="bar"/>
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#postMethod" />
+              </resource>
+           </resources>
+           <method id="postMethod" name="POST">
+               <request>
+                 <representation mediaType="application/json">
+                 </representation>
+               </request>
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (header single)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="id" style="header" repeating="false" required="true"/>
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#postMethod" />
+              </resource>
+           </resources>
+           <method id="postMethod" name="POST">
+               <request>
+                 <representation mediaType="application/json">
+                 </representation>
+               </request>
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
+    scenario ("Given a transform with Rax-Roles on a teant but mismatch in case of tenant parameter (capture header)") {
+      Given("a WADL which leverages rax:roles when there is no tenant match")
+      val inWADL = <application xmlns="http://wadl.dev.java.net/2009/02"
+                     xmlns:rax="http://docs.rackspace.com/api"
+                     xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+           <grammars/>
+           <resources base="https://test.api.openstack.com">
+              <resource path="path/to/my/resource/{stepType}" rax:roles="admin/{ID}">
+                   <param name="stepType" style="template" type="xsd:string"/>
+                   <method href="#postMethod" />
+              </resource>
+           </resources>
+           <method id="postMethod" name="POST">
+               <request>
+                <representation mediaType="application/json">
+                  <rax:captureHeader name="id" path="//@id"/>
+                 </representation>
+               </request>
+               <response status="200 203"/>
+           </method>
+      </application>
+      When("The WADL is translated")
+      val checkerLog = log(Level.ERROR) {
+        intercept[WADLException] {
+          val checker = builder.build(inWADL, raxRolesConfig)
+          println (checker) // Should never print!
+        }
+      }
+      Then ("There should be an error detailing that rax:roles mask is currently not supported")
+      assert(checkerLog, "no defined param named 'ID'")
+    }
+
 
     scenario ("Given a transform with Rax-Roles on a teant with a role name that contains a \\") {
       Given("a WADL which leverages rax:roles tenanted with a role name that contains a \\")

